test/Microsoft.SqlTools.ServiceLayer.UnitTests/EditData/SessionTests.cs

```diff
@@ -253,26 +278,260 @@ public static IEnumerable<object> RowIdOutOfRangeData
+            EditSession s = new EditSession(emf.Object, Constants.OwnerUri, Constants.OwnerUri);
+
+            // ... Create a mock for verifying the failure handler will be called
+            var successHandler = DoNothingSuccessMock;
+            var failureHandler = DoNothingFailureMock;
+
+            // If: I initalize the session with a metadata factory that will fail
+            s.Initialize(DoNothingConnector, DoNothingQueryRunner, successHandler.Object, failureHandler.Object);
+            await s.InitializeTask;
+
+            // Then:
+            // ... The session should not be initialized
+            Assert.False(s.IsInitialized);
+
+            // ... The failure handler should have been called once
+            failureHandler.Verify(f => f(It.IsAny<Exception>()), Times.Once);
+
+            // ... The success handler should not have been called at all
+            successHandler.Verify(f => f(), Times.Never);
+        }
+
+        [Fact]
+        public async Task InitializeQueryFailException()
+        {
+            // Setup:
+            // ... Create a metadata factory that will return some generic column information
+            var b = QueryExecution.Common.GetBasicExecutedBatch();
+            var etm = Common.GetStandardMetadata(b.ResultSets[0].Columns);
+            Mock<IEditMetadataFactory> emf = new Mock<IEditMetadataFactory>();
+            emf.Setup(f => f.GetObjectMetadata(It.IsAny<DbConnection>(), It.IsAny<string>(), It.IsAny<string>()))
+                .Returns(etm);
+
+            // ... Create a session that hasn't been initialized
+            EditSession s = new EditSession(emf.Object, Constants.OwnerUri, Constants.OwnerUri);
+
+            // ... Create a query runner that will fail via exception
+            Mock<EditSession.QueryRunner> qr = new Mock<EditSession.QueryRunner>();
+            qr.Setup(r => r(It.IsAny<string>())).Throws(new Exception("qqq"));
+
+            // ... Create a mock for verifying the failure handler will be called
+            var successHandler = DoNothingSuccessMock;
+            var failureHandler = DoNothingFailureMock;
+
+            // If: I initialize the session with a query runner that will fail
+            s.Initialize(DoNothingConnector, qr.Object, successHandler.Object, failureHandler.Object);
+            await s.InitializeTask;
+
+            // Then:
+            // ... The session should not be initialized
+            Assert.False(s.IsInitialized);
+
+            // ... The failure handler should have been called once
+            failureHandler.Verify(f => f(It.IsAny<Exception>()), Times.Once);
+
+            // ... The success handler should not have been called at all
+            successHandler.Verify(f => f(), Times.Never);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("It fail.")]
+        public async Task InitializeQueryFailReturnNull(string message)
+        {
+            // Setup:
```

Comment:
Minor: it seems like overall setup for these methods is nearly identical. Might a single private Setup method save some space and tighten up the test code?